datafusion/physical-plan/src/aggregates/group_values/multi_group_by/primitive.rs

```diff
@@ -35,6 +36,7 @@ use std::sync::Arc;
 /// `NULLABLE`: if the data can contain any nulls
 #[derive(Debug)]
 pub struct PrimitiveGroupValueBuilder<T: ArrowPrimitiveType, const NULLABLE: bool> {
+    data_type: DataType,
```

Comment:
I double checked and the single group by column implementation does the same thing
https://github.com/apache/datafusion/blob/c0ca4b4e449e07c3bcd6f3593fa31dd31ed5e0c5/datafusion/physical-plan/src/aggregates/group_values/single_group_by/primitive.rs#L84

👍